src/identity/interaction/email-password/credentials/DeleteCredentialsHandler.ts

```diff
@@ -0,0 +1,45 @@
+import type { Adapter } from 'oidc-provider';
+import { BasicRepresentation } from '../../../../http/representation/BasicRepresentation';
+import type { Representation } from '../../../../http/representation/Representation';
+import { APPLICATION_JSON } from '../../../../util/ContentTypes';
+import { BadRequestHttpError } from '../../../../util/errors/BadRequestHttpError';
+import { NotImplementedHttpError } from '../../../../util/errors/NotImplementedHttpError';
+import type { AdapterFactory } from '../../../storage/AdapterFactory';
+import type { AccountStore } from '../storage/AccountStore';
+import type { CredentialsHandlerInput } from './CredentialsHandler';
+import { CredentialsHandler } from './CredentialsHandler';
+
+/**
+ * Handles the deletion of credential tokens.
+ * Expects the JSON body to have a `delete` field with as value the ID of the token to be deleted.
+ */
+export class DeleteCredentialsHandler extends CredentialsHandler {
+  private readonly adapter: Adapter;
+  private readonly accountStore: AccountStore;
+
+  public constructor(factory: AdapterFactory, accountStore: AccountStore) {
+    super();
+    this.adapter = factory.createStorageAdapter('Client');
+    this.accountStore = accountStore;
+  }
+
+  public async canHandle({ body }: CredentialsHandlerInput): Promise<void> {
+    if (typeof body.delete !== 'string') {
+      throw new NotImplementedHttpError();
+    }
+  }
+
+  public async handle({ operation, body }: CredentialsHandlerInput): Promise<Representation> {
+    const id = body.delete as string;
+    const settings = await this.accountStore.getSettings(body.webId);
+    const idx = settings.clientCredentials.indexOf(id);
+    if (idx < 0) {
+      throw new BadRequestHttpError(`${id} is not assigned to this account.`);
```

Comment:
```suggestion
      throw new BadRequestHttpError('No credential with this ID exists for this account.');
```
You can log the warning, but let's not emit sensitive info to clients.

Reply:
This is an ID we just received in a POST request, meaning the user sent it to us.

Reply:
Even echoing is not okay; but per the below statement that it is not actually a secret, that's good then.